clients/cloud/rust/src/producer.rs

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+use futures::future::Future;
+use rdkafka::error::KafkaError;
+use rdkafka::message::OwnedMessage;
+use rdkafka::producer::{FutureProducer, FutureRecord};
+use std::boxed::Box;
+
+mod utils;
+
+fn log_produce_result(
+    topic: &str,
+    result: Result<(i32, i64), (KafkaError, OwnedMessage)>,
+) -> Result<(), ()> {
```

Comment:
If there is nothing to return for the result, do we even need to return anything?

Reply:
I need to if I want to use it in an `and_then` like I do